File: frontend/projects/irrtirer/src/app/shared/mosaic-generator/sectors/sector.model.ts
import { Vector } from '../../../core/models/vector.model';
import { TileTransform } from '../models/tile-transform.model';
import { EvaluationParams } from './generation-params/evaluation-params.interface';
import { PopulationParams } from './generation-params/population-params.interface';
import { SectionModel } from './section.model';

export class SectorModel {
    public readonly id: string;

    public tileMinRadius: number;
    public tileMaxRadius: number;
    public tileMargin: number;

    public evaluationParams: EvaluationParams;
    public populationParams: PopulationParams;

    public contour: Vector[];

    public sections: SectionModel[] = [];

    constructor(id: string) {
        this.id = id;
    }

    public getSectorTiles(): TileTransform[] {
        return this.sections
            .filter(section => section.wasGenerated)
            .flatMap(section => section.directTiles);
    }

    public static restore(obj: SectorModel): void {
        Object.setPrototypeOf(obj, SectorModel.prototype);
        for(const section of obj.sections) {
            SectionModel.restore(section, obj);
        }
    }
}

File: backend/Irrtirer/Irrtirer.Generator/MosaicStructure/SectorModel.cs
using Irrtirer.Generator.Models.GenerationParams;
using System.Collections.Generic;
using System.Linq;
using System.Numerics;

namespace Irrtirer.Generator.MosaicStructure
{
    public class SectorModel
    {
        public float TileMinRadius { get; set; }
        public float TileMaxRadius { get; set; }
        public float TileMargin { get; set; }

        public EvaluationParams EvaluationParams { get; set; }
        public PopulationParams PopulationParams { get; set; }

        public Vector2[] Contour { get; set; }

        public SectionModel[] Sections { get; set; }

        internal IEnumerable<TileTransform> SectorTiles
        {
            get
            {
                return Sections
                    .Where(section => section.WasGenerated)
                    .SelectMany(section => section.DirectTiles);
            }
        }
    }
}
